DPPP/test/unit/tBDABuffer.cc

```diff
@@ -0,0 +1,391 @@
+    BOOST_CHECK_EQUAL(*buffer.GetWeights(), new_weights[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetWeights() + 1), new_weights[1]);
+    BOOST_CHECK_EQUAL(*(buffer.GetWeights() + 2), new_weights[2]);
+}
+
+BOOST_AUTO_TEST_CASE( test_modify_full_resflags )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {4};
+    BDABuffer buffer {n_channels * n_correlations};
+    bool flags[] {false, true, false};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, nullptr, flags);
+    
+    bool new_flags[] {true, false, true};
+    *buffer.GetFullResFlags() = new_flags[0];
+    *(buffer.GetFullResFlags() + 1) = new_flags[1];
+    *(buffer.GetFullResFlags() + 2) = new_flags[2];
+    BOOST_CHECK_EQUAL(*buffer.GetFullResFlags(), new_flags[0]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFullResFlags() + 1), new_flags[1]);
+    BOOST_CHECK_EQUAL(*(buffer.GetFullResFlags() + 2), new_flags[2]);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_less ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsLess(5., 10.), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_less_false ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsLess(15., 10.), false);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_less_near ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsLess(9.99999998, 10.0000000), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_less_near_false ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsLess(9.99999999, 10.0000000), false);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_greater_false ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsGreaterEqual(5., 10.), false);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_greater ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsGreaterEqual(15., 10.), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_greater_near_false ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsGreaterEqual(9.99999998, 10.0000000), false);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_greater_near ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsGreaterEqual(9.999999999, 10.0000000), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_greater_equal ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsGreaterEqual(10.0000000, 10.0000000), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_equal ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsEqual(10.0000000, 10.0000000), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_not_equal ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsEqual(20.0000000, 10.0000000), false);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_near_equal ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsEqual(9.999999991, 10.0000000), true);
+}
+
+BOOST_AUTO_TEST_CASE( test_time_is_not_near_equal ) {
+    BDABuffer buffer {0};
+    BOOST_CHECK_EQUAL(buffer.TimeIsEqual(9.99999999, 10.0000000), false);
+}
```

Comment:
The Time* functions are static, so you don't need to create a BDABuffer in these tests.
Simply BDABuffer::TimeIsEqual(...) also works.

Also: Personally I prefer BOOST_CHECK(X) over BOOST_CHECK_EQUAL(X, true) and BOOST_CHECK(!X) over BOOST_CHECK_EQUAL(X, false).